### DISCUSSION TITLE
Feature Suggestions for Screenbox Media Player

### DISCUSSION THREAD

**Discussion by Al-kides** [huynhsontung/Screenbox#463] [General]
Hello,

I hope this message finds you well. I have a couple of questions and suggestions regarding the Screenbox Media Player.
Linux Compatibility: Will this software be available for Linux users in the future?
Sorting Options: It would be great to have a sorting feature similar to that in Windows Media Player, allowing users to change the order of videos based on selected criteria.
![image](https://github.com/user-attachments/assets/ab49fbaf-9792-4fe8-bbbc-3bbc203bc647)
Video Categorization: I suggest adding a section to categorize videos. For instance, users could have separate folders for Movies, TV Shows, and General Videos, similar to how PowerDVD organizes content.
![image](https://github.com/user-attachments/assets/5f77d1fa-1972-44ad-99ba-396a7c118d16)

Layout Improvement: To enhance user experience, perhaps consider a layout inspired by Da Vinci Resolve, where different media options are clearly outlined at the bottom. This could improve the distribution of videos and music on the screen.
![image](https://github.com/user-attachments/assets/5fb1e215-9778-4e6e-aded-f6fa0caaab7f)

Thanks, 
A newbie dev who wants to contribute someday but can't because of skill issues

**Comment by United600** [comment]
> Linux Compatibility: Will this software be available for Linux users in the future?

Unlikely.

> Video Categorization: I suggest adding a section to categorize videos. For instance, users could have separate folders for Movies, TV Shows, and General Videos, similar to how PowerDVD organizes content.

I don't know how PowerDVD determines the category but auto categorization also seems unlikely, why not organize with folders instead?

> Layout Improvement: To enhance user experience, perhaps consider a layout inspired by Da Vinci Resolve, where different media options are clearly outlined at the bottom. This could improve the distribution of videos and music on the screen.

If I'm understanding correctly you want a clear separation of tracks? That's typically a feature of editors, not players. Or do you mean a media filter?

**Reply by Al-kides** [reply]
1. Oh well, I understand it would be a lot of work for little returns.
2. Folder organization is a good solution but looking at it, I'm worried about the future in that you might want add "movie" specific features, for example. Given I'm not too experience in media programming, I am unsure about what that would entail, unfortunately. I feel a little silly for even suggesting it.
3. This was to do with request 2, where you'll have movies having it's own section (making it possible and convenient if you were to decide one day to fetch metadata from sites like imndb), series (same thing) and just random videos from random sites, as I already pretty much described in point two.

Thanks for taking time out of your day to respond 🙏🏻

**Reply by United600** [reply]
All suggestions are welcomed. Yeah metadata fetching might come at some point in the future, enabling auto-categorization, but don't hold me to that as is not set in stone.

**Comment by huynhsontung** [comment]
Just to add to @United600 answer. Linux support is unlikely because the entire UI stack is based on Windows technologies. This app is purposely built just for the Windows platform. It takes advantage of the app model of Windows, UI style and rendering of Windows, native input handling and accessibility features of Windows. Porting the app to Linux would lose all of the features that made people like Screenbox.

**Reply by Al-kides** [reply]
I understand, thank you 🙏🏻

**Reply by os-guy** [reply]
I don't know exactly what interface library it uses, but there is an app on Flathub that looks exactly like the Windows 11 interface.
https://flathub.org/apps/me.timschneeberger.GalaxyBudsClient

I recommend you to take a look at this.

EDIT: I think it uses a framework, called [AvaloniaUI](https://avaloniaui.net/)

**Reply by United600** [reply]
Without some major refactoring, using Avalonia doesn't seem realistic. Uno would be relatively easier (and a closer fit), but again I don't think, as of right now, it's realistically feasible and we would lose Xbox support and the benefits of pure UWP.
